index.bs

```diff
@@ -560,79 +560,251 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
+				constructor(record&lt;DOMString, ClipboardItemData> items,
+				            optional ClipboardItemOptions options = {});
 
-		typedef (DOMString or Blob) ClipboardItemDataType;
-		typedef Promise&lt;ClipboardItemDataType> ClipboardItemData;
+				readonly attribute PresentationStyle presentationStyle;
+				readonly attribute FrozenArray&lt;DOMString> types;
 
-		callback ClipboardItemDelayedCallback = ClipboardItemData ();
+				Promise&lt;Blob&gt; getType(DOMString type);
+			};
 
-		[Exposed=Window] interface ClipboardItem {
-		  constructor(record&lt;DOMString, ClipboardItemData> items,
-		    optional ClipboardItemOptions options = {});
-		  static ClipboardItem createDelayed(
-		      record&lt;DOMString, ClipboardItemDelayedCallback> items,
-		      optional ClipboardItemOptions options = {});
+			enum PresentationStyle { "unspecified", "inline", "attachment" };
 
-		  readonly attribute PresentationStyle presentationStyle;
-		  readonly attribute long long lastModified;
-		  readonly attribute boolean delayed;
+			dictionary ClipboardItemOptions {
+				PresentationStyle presentationStyle = "unspecified";
+			};
+		</pre>
 
-		  readonly attribute FrozenArray&lt;DOMString> types;
+		<dl class=note>
+			<dt><code><var>clipboardItem</var> = new ClipboardItem([<var>items</var>, <var>options</var>])</code>
+			<dd>
+			Creates a new {{ClipboardItem}} object. <var>items</var> are used to fill its MIME types and [=Promise=]s to {{Blob}}s or {{DOMString}}s corresponding to the MIME types, <var>options</var> can be used to fill its {{ClipboardItemOptions}},
+			as per the example below.
 
-		  Promise&lt;Blob> getType(DOMString type);
-		};
+				<pre class="example javascript" highlight=js>
+					const format1 = 'text/plain';
+					const promise_text_blob = Promise.resolve(new Blob(['hello'], {type: format1}));
+					const clipboardItemInput = new ClipboardItem(
+						{[format1]: promise_text_blob},
+						{presentationStyle: "unspecified"});
+				</pre>
+
+			<dt><code><var>clipboardItem</var>.getType(<var>type</var>)</code>
+			<dd><p>Returns a [=Promise=] to the {{Blob}} corresponding to <var>type</var>.</p>
+
+			<dt><code><var>clipboardItem</var>.<var>types</var></code>
+			<dd><p>Returns the list of <var>types</var> contained in the <var>clipboardItem</var> object.
+
+		</dl>
+
+		<h4 id="clipboard-item">Clipboard Item</h4>
+
+		A <dfn>clipboard item</dfn> is conceptually data that the user has expressed a desire to make shareable by invoking a "cut" or "copy" command.
+
+		<p class=note>
+			For example, if a user copies a range of cells from a spreadsheet of a native application, it will result in one [=/clipboard item=]. If a user copies a set of files from their desktop, that list of files will be represented by multiple [=/clipboard item=]s.
+		</p>
+
+		Some platforms may support having more than one [=/clipboard item=] at a time on the [=clipboard=], while other platforms replace the previous [=/clipboard item=] with the new one.
```

Comment:
This is vaguely formulated, because it's unclear which "previous clipboard item" is meant.

How about moving this as a note to the `read()` section? After step 3.3. ("Let data be a copy of the system clipboard data represented as clipboard items.[...]). Ideally, that platform-specific part would have to be specified in detail for each platform, however in order to limit the scope of this review, I suggest adding it as a note. CC @domenic 
For `write()` a note about the platform-dependency would have to be added too, after step 3.3.3. Something like
"As the ClipboardItems are written in order, this implies for platforms supporting only exactly one ClipboardItem, the last item replaces the previously written ones."

Reply:
In `write()`, the algorithm has a step to deal with multiple `clipboard item`s: 
`1. If |data|'s [=list/size=] is greater than 1, and the current operating system does not support multiple [=/clipboard item=]s on the [=system clipboard data=], then set |data| to |data|[0].`

Reply:
Thanks for the clarification. So to be strictly correct `data[0]` can't be assigned to `data`. The former represents one `ClipboardItem` object, the latter a `sequence`. The next step of the algorithm is broken because of that.
Either `data` has to be correctly truncated to the first element or it needs to be replaced by a new sequence containing only the first element.

Reply:
Yes, that is correct that `data[0]` can't be assigned to `data`. I'll change it. Thanks!